tests/models/test_image.py

```diff
@@ -279,6 +288,101 @@ def test_image_coordinates():
+                            ra00=119.90210641, ra01=119.90210641, ra10=120.16318787, ra11=120.16318787,
+                            dec00=39.975, dec01=40.125, dec10=39.975, dec11=40.125,
+                            provenance=provenance_base, nofile=True, **kwargs )
+            # imagepoint and imagefar are used to test Image.containing and Image.find_containing,
+            # as Image is the only example of a SpatiallyIndexed thing we have so far.
+            # The corners don't matter for these given how they'll be used.
+            imagepoint = Image( 'point.fits', ra=119.88, dec=39.95, 
+                                ra00=-.001, ra01=0.001, ra10=-0.001, ra11=0.001, dec00=0, dec01=0, dec10=0, dec11=0,
+                                provenance=provenance_base, nofile=True, **kwargs )
+            imagefar = Image( 'far.fits', ra=30, dec=-10, 
+                                ra00=0, ra01=0, ra10=0, ra11=0, dec00=0, dec01=0, dec10=0, dec11=0,
+                                provenance=provenance_base, nofile=True, **kwargs )
+            session.add( image1 )
+            session.add( image2 )
+            session.add( image3 )
+            session.add( imagepoint )
+            session.add( imagefar )
+
+            sought = session.query( Image ).filter( Image.containing( 120, 40 ) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'one.fits', 'two.fits', 'three.fits' } )
+
+            sought = session.query( Image ).filter( Image.containing( 119.88, 39.95 ) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'one.fits' } )
+
+            sought = session.query( Image ).filter( Image.containing( 120, 40.12 ) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'two.fits', 'three.fits' } )
+
+            sought = session.query( Image ).filter( Image.containing( 120, 39.88 ) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'two.fits' } )
+
+            sought = Image.find_containing( imagepoint, session=session )
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'one.fits' } )
+
+            sought = session.query( Image ).filter( Image.containing( 0, 0 ) ).all()
+            assert len(sought) == 0
+
+            sought = Image.find_containing( imagefar, session=session )
+            assert len(sought) == 0
+
+            sought = session.query( Image ).filter( Image.within( image1 ) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'one.fits', 'two.fits', 'three.fits', 'point.fits' } )
+
+            sought = session.query( Image ).filter( Image.within( imagefar ) ).all()
+            assert len(sought) == 0
```

Comment:
Here I would also do a `not in` test against the image ids, instead of assuming that there are not other images in the database.